tracing/trace.go

```diff
@@ -66,7 +66,7 @@ func (t *trace) addCounter(key string, delta float64) {
 
 func (t *trace) setTag(key string, value interface{}) {
 	if v, ok := value.(bool); ok {
-		value = interface{}(strconv.FormatBool(v))
+		value = strconv.FormatBool(v)
```

Comment:
🔕 Actually I wonder if we should just convert anything that's not string into string? e.g.

```
if _, ok := value.(string); !ok {
  value = fmt.Sprintf("%v", v)
}
```

Or maybe change `value` from `interface{}` into `string`?

Reply:
🤔 

I think converting everything to a string makes sense, since this is a detail of the current system we use for tracing rather than a part of the tracing spec, I don't think we want to encode that into the API.

We could make the type on `setTag` into string and do the conversion in `Span.SetTag` though?